chat-widget/src/components/webchatcontainerstateful/webchatcontroller/markdownrenderers/HyperlinkTextOverrideRenderer.ts

```diff
@@ -0,0 +1,63 @@
+import DOMPurify from "dompurify";
+
+class HyperlinkTextOverrideRenderer {
+    private hyperlinkTextOverride: boolean
+    constructor(hyperlinkTextOverride: boolean) {
+        this.hyperlinkTextOverride = hyperlinkTextOverride;
+    }
+
+    private convertTextToHtmlNode(text: string) {
+        const htmlNode = document.createElement("div");
+
+        try {
+            text = DOMPurify.sanitize(text); // eslint-disable-line @typescript-eslint/no-explicit-any
+            htmlNode.innerHTML = text;
+        } catch {
+            return htmlNode;
+        }
+
+        return htmlNode;
+    }
+
+    private processANode(htmlNode: any) { // eslint-disable-line @typescript-eslint/no-explicit-any
+        const aTags = htmlNode.getElementsByTagName("a");
+
+        for (let index = 0; index < aTags.length; index++) {
+            const aNode = aTags[index];
+
+            if (!aNode ||
+                !aNode.tagName || aNode.tagName.toLowerCase() !== "a" ||
+                !aNode.href) continue;
+            if (aNode.href !== aNode.innerText.trim()) {

```

Comment:
null check on aNode.innerText

Reply:
It shouldn't be needed since `htmlNode.getElementsByTagName("a")` would always return a `HTMLCollection` of anchor elements. An anchor element would always have `innerText`. If the `HTMLCollection` would return nothing, then collection would always have a size of `0`, then the loop would be ignored.